### ISSUE TITLE
Histogram aggregation on range fields using lte/lt or gte/gt

### ISSUE THREAD

**Issue by tomasfalt** [elastic/elasticsearch#23414]
Since it's impossible to do a histogram aggregation directly on a range field (see #23182) shouldn't it be possible to do it on any of the lte/lt or gte/gt fields?

**Comment by jpountz** [comment]
Is it a feature request to be able to aggegate on range bounds, or is it a request for a workaround until #23182 is implemented?

**Comment by tomasfalt** [comment]
Both actually. To start with as a workaround but I think there are several scenarios which will benefit from aggregation on the range bounds, for example the calculation of the churn rate for a certain intervals as a basis for churn prediction.

**Comment by tomasfalt** [comment]
Also, it would be great if the max and min aggregations (and probably all other metrics aggregations)  would work on the range bounds.

**Comment by jpountz** [comment]
Agreed, there are several aggregations that could be generalized on range fields. We just started looking into doc values support for range fields, so I would not expect seeing the feature out in the short term however.

**Comment by polyfractal** [comment]
Should be largely unblocked at the moment (range fields have doc values, they are enabled by default now as of https://github.com/elastic/elasticsearch/pull/24823, etc), and I think it's a feature we'd like to support eventually.

Adding `adoptme` tag though, since I don't think it is on anyone's immediate plate of work.

Note: the error message from trying to aggregate on a range_field is rather unhelpful since it tries to use field data (even though doc values are enabled):

```
{
  "type": "illegal_argument_exception",
  "reason": "Fielddata is not supported on field [expected_attendees] of type [integer_range]"
}
```

**Comment by polyfractal** [comment]
Closing in favor of #34644
